### PR TITLE
Added GoboRotatingMacro preset and ActsOn attribute to first 28 Fixtures

### PR DESCRIPTION
**Pull request by kluszon**
Implementation of changes of for better control of Gobos as agreed in https://www.qlcplus.org/forum/viewtopic.php?f=35&t=14353. Rotation presets have also been added where missing.

(cherry picked from commit addbee6165f2400a948d1858f22dbbf28c2e97ba)

### CODE REVIEW

**Comment by kluszon** [comment]
@mcallegari Can you review my changes?

File: resources/fixtures/American_DJ/American-DJ-Focus-Spot-Three-Z.qxf [hunk 1]
```diff
@@ -44,36 +44,36 @@
...
   <Group Byte="0">Gobo</Group>
   <Capability Min="0" Max="9" Preset="GoboMacro" Res1="Others/open.svg">Open</Capability>
-  <Capability Min="10" Max="18" Preset="GoboMacro" Res1="ClayPaky/gobo00052.svg">Gobo 1</Capability>
-  <Capability Min="19" Max="27" Preset="GoboMacro" Res1="Others/gobo00025.svg">Gobo 2</Capability>
-  <Capability Min="28" Max="36" Preset="GoboMacro" Res1="Chauvet/gobo00049.svg">Gobo 3</Capability>
-  <Capability Min="37" Max="45" Preset="GoboMacro" Res1="SGM/gobo00058.png">Gobo 4</Capability>
-  <Capability Min="46" Max="54" Preset="GoboMacro" Res1="SGM/gobo00027.svg">Gobo 5</Capability>
-  <Capability Min="55" Max="63" Preset="GoboMacro" Res1="SGM/gobo00035.png">Gobo 6</Capability>
-  <Capability Min="64" Max="73" Preset="GoboMacro" Res1="Others/open.svg">Open</Capability>
+  <Capability Min="10" Max="18" Preset="GoboRotatingMacro" Res1="ClayPaky/gobo00052.svg">Gobo 1</Capability>
```
Comment (mcallegari) [inline resources/fixtures/American_DJ/American-DJ-Focus-Spot-Three-Z.qxf:54]: Keep it GoboMacro

File: resources/fixtures/American_DJ/American-DJ-Inno-Pocket-Spot-Twins.qxf [hunk 2]
```diff
@@ -14,24 +14,24 @@
...
   <Group Byte="0">Gobo</Group>
   <Capability Min="0" Max="7" Preset="GoboMacro" Res1="Others/open.svg">Open</Capability>
-  <Capability Min="8" Max="15">Gobo 1</Capability>
-  <Capability Min="16" Max="23">Gobo 2</Capability>
-  <Capability Min="24" Max="31">Gobo 3</Capability>
-  <Capability Min="32" Max="39">Gobo 4</Capability>
-  <Capability Min="40" Max="47">Gobo 5</Capability>
-  <Capability Min="48" Max="55">Gobo 6</Capability>
-  <Capability Min="56" Max="63">Gobo 7</Capability>
+  <Capability Min="8" Max="15" Preset="GoboMacro" Res1="">Gobo 1</Capability>
```
Comment (mcallegari) [inline resources/fixtures/American_DJ/American-DJ-Inno-Pocket-Spot-Twins.qxf:17]: Without a picture, a preset alone doesn't make much sense for preview

**Review by mcallegari — CHANGES_REQUESTED** [review]
Fix useless GoboRotatingMacro ones.
ALL of them, not just the ones I commented...I got bored after a while

File: resources/fixtures/American_DJ/American-DJ-Accu-Spot-250-II.qxf [hunk 3]
```diff
@@ -30,28 +30,28 @@
...
   <Group Byte="0">Gobo</Group>
   <Capability Min="0" Max="9" Preset="GoboMacro" Res1="Others/open.svg">Spot</Capability>
-  <Capability Min="10" Max="19" Preset="GoboMacro" Res1="Others/gobo00036.svg">Star Circle (Metal)</Capability>
-  <Capability Min="20" Max="29" Preset="GoboMacro" Res1="Others/gobo00023.svg">Cross (Metal)</Capability>
-  <Capability Min="30" Max="39" Preset="GoboMacro" Res1="Others/gobo00035.png">Blue &amp; White Web (Dichoric)</Capability>
-  <Capability Min="40" Max="49" Preset="GoboMacro" Res1="Others/gobo00034.png">Textured Glass (Glass)</Capability>
-  <Capability Min="50" Max="59" Preset="GoboMacro" Res1="Others/gobo00037.svg">Donut (Metal)</Capability>
-  <Capability Min="60" Max="69" Preset="GoboMacro" Res1="Others/gobo00027.svg">Triangle (Metal)</Capability>
-  <Capability Min="70" Max="79" Preset="GoboMacro" Res1="Others/gobo00029.svg">Dot Line (Metal)</Capability>
+  <Capability Min="10" Max="19" Preset="GoboRotatingMacro" Res1="Others/gobo00036.svg">Star Circle (Metal)</Capability>
```
[the remark this answers is not in the document]
Comment (joshuaurieli) [reply resources/fixtures/American_DJ/American-DJ-Accu-Spot-250-II.qxf:33]: Why? This device has one gobo group and one gobo rotation group, so the gobo group must contain rotating gobos. Furthermore, the manual clearly states these gobos are rotating gobos: https://cdb.s3.amazonaws.com/ItemRelatedFiles/7404/accu_spot_250_ll.pdf

File: resources/fixtures/American_DJ/American-DJ-Accu-Spot-Pro.qxf [hunk 4]
```diff
@@ -101,7 +101,7 @@
   <Capability Min="32" Max="63" Preset="ShutterOpen">Shutter open</Capability>
   <Capability Min="64" Max="95" Preset="StrobeSlowToFast">Strobe Effect Slow - Fast</Capability>
   <Capability Min="96" Max="127" Preset="ShutterOpen">Shutter open</Capability>
-  <Capability Min="128" Max="159" Preset="PulseFrequency">Pulse Effect</Capability>
+  <Capability Min="128" Max="159" Preset="PulseFrequency" Res1="0">Pulse Effect</Capability>
```
[the remark this answers is not in the document]
Comment (joshuaurieli) [reply resources/fixtures/American_DJ/American-DJ-Accu-Spot-Pro.qxf:104]: This is a default added by the editor. The current version of the editor requires a Res1 for PulseFrequency. If there was no Res1 in the previous version of the file, the default is added. We left it, assuming the editor is correct.

File: resources/fixtures/American_DJ/American-DJ-Auto-Spot-150.qxf [hunk 5]
```diff
@@ -14,22 +14,22 @@
...
   <Capability Min="0" Max="15" Preset="GoboMacro" Res1="Others/open.svg">Open</Capability>
-  <Capability Min="16" Max="31" Preset="GoboMacro" Res1="SGM/gobo00007.png">Gobo 1</Capability>
-  <Capability Min="32" Max="47" Preset="GoboMacro" Res1="Robe/gobo00035.svg">Gobo 2</Capability>
-  <Capability Min="48" Max="63" Preset="GoboMacro" Res1="Robe/gobo00038.svg">Gobo 3</Capability>
-  <Capability Min="64" Max="79" Preset="GoboMacro" Res1="Others/gobo00005.svg">Gobo 4</Capability>
-  <Capability Min="80" Max="95" Preset="GoboMacro" Res1="Others/gobo00032.svg">Gobo 5</Capability>
-  <Capability Min="96" Max="111" Preset="GoboMacro" Res1="Robe/gobo00037.svg">Gobo 6</Capability>
-  <Capability Min="112" Max="127" Preset="GoboMacro" Res1="SGM/gobo00027.svg">Gobo 7</Capability>
-  <Capability Min="128" Max="255" Preset="GenericPicture" Res1="Others/rainbow.png">Rotating slow-fast</Capability>
+  <Capability Min="16" Max="31" Preset="GoboRotatingMacro" Res1="SGM/gobo00007.png">Gobo 1</Capability>
```
[the remark this answers is not in the document]
Comment (joshuaurieli) [reply resources/fixtures/American_DJ/American-DJ-Auto-Spot-150.qxf:17]: See my previous comment.

File: resources/fixtures/American_DJ/American-DJ-Focus-Spot-4Z.qxf [hunk 6]
```diff
@@ -30,12 +30,12 @@
  <Channel Name="Gobo Wheel">
   <Group Byte="0">Gobo</Group>
   <Capability Min="0" Max="4" Preset="GoboMacro" Res1="Others/open.svg">Open</Capability>
-  <Capability Min="5" Max="19" Preset="GoboMacro" Res1="Others/gobo00013.svg">Gobo 1</Capability>
-  <Capability Min="20" Max="34" Preset="GoboMacro" Res1="Others/gobo00115.svg">Gobo 2</Capability>
-  <Capability Min="35" Max="49" Preset="GoboMacro" Res1="SGM/gobo00027.svg">Gobo 3</Capability>
-  <Capability Min="50" Max="64" Preset="GoboMacro" Res1="Others/gobo00023.svg">Gobo 4</Capability>
-  <Capability Min="65" Max="79" Preset="GoboMacro" Res1="Others/gobo00086.svg">Gobo 5</Capability>
-  <Capability Min="80" Max="94" Preset="GoboMacro" Res1="Others/gobo00008.svg">Gobo 6</Capability>
+  <Capability Min="5" Max="19" Preset="GoboRotatingMacro" Res1="Others/gobo00013.svg">Gobo 1</Capability>
```
[the remark this answers is not in the document]
Comment (joshuaurieli) [reply resources/fixtures/American_DJ/American-DJ-Focus-Spot-4Z.qxf:33]: See my previous comment.

File: resources/fixtures/American_DJ/American-DJ-Focus-Spot-Three-Z.qxf [hunk 7]
```diff
@@ -24,7 +24,7 @@
   <Capability Min="57" Max="63" Preset="ColorMacro" Res1="#ff55ff">Pink</Capability>
   <Capability Min="64" Max="127" Preset="GoboMacro" Res1="Others/rainbow.png">Color mixing</Capability>
   <Capability Min="128" Max="189" Preset="GoboMacro" Res1="Others/rainbow.png">Counter clockwise rotation fast - slow</Capability>
```
[the remark this answers is not in the document]
Comment (joshuaurieli) [reply resources/fixtures/American_DJ/American-DJ-Focus-Spot-Three-Z.qxf:26]: OK. What should be the general rule here? Change all GoboMacro presets with Res1="Others/rainbow.png" to a Rotation preset? Or only those with some mention of rotation in the description?

File: resources/fixtures/American_DJ/American-DJ-Inno-Pocket-Spot-Twins.qxf (hunk 2 above)
Comment (joshuaurieli) [reply resources/fixtures/American_DJ/American-DJ-Inno-Pocket-Spot-Twins.qxf:17]: This is the agreed approach: https://www.qlcplus.org/forum/viewtopic.php?f=35&t=14353&start=10#p60205

File: resources/fixtures/American_DJ/American-DJ-Accu-Spot-250-II.qxf (hunk 3 above)
[the remark this answers is not in the document]
Comment (mcallegari) [reply resources/fixtures/American_DJ/American-DJ-Accu-Spot-250-II.qxf:33]: Then the same rule should apply to the shaking gobos. They should be "GoboRotatingShakeMacro"
In principle, a gobo wheel cannot mechanically be a mixed case (rotating + non rotating)
So I would keep simple "GogoMacro" and then apply the rotation with the "ActsOn" feature

File: resources/fixtures/American_DJ/American-DJ-Accu-Spot-Pro.qxf (hunk 4 above)
[the remark this answers is not in the document]
Comment (mcallegari) [reply resources/fixtures/American_DJ/American-DJ-Accu-Spot-Pro.qxf:104]: "PulseFrequency" means a specific frequency, which cannot be 0.
You probably meant a generic one like PulseSlowToFast or PulseFastToSlow

File: resources/fixtures/American_DJ/American-DJ-Vizi-Roller-Beam-2R.qxf [hunk 8]
```diff
@@ -121,7 +118,7 @@
   <Capability Min="32" Max="63" Preset="ShutterOpen">Shutter Open</Capability>
   <Capability Min="64" Max="95" Preset="StrobeSlowToFast">Strobing Slow to Fast</Capability>
   <Capability Min="96" Max="127" Preset="ShutterOpen">Shutter Open</Capability>
-  <Capability Min="128" Max="159" Preset="PulseFrequency">Pulse Effect in Sequences</Capability>
+  <Capability Min="128" Max="159" Preset="PulseFrequency" Res1="0">Pulse Effect in Sequences</Capability>
```
Comment (mcallegari) [inline resources/fixtures/American_DJ/American-DJ-Vizi-Roller-Beam-2R.qxf:121]: Use a generic preset

File: resources/fixtures/Briteq/Briteq-BT-575S.qxf [hunk 9]
```diff
@@ -166,29 +166,14 @@
   <Channel Number="6">No function</Channel>
   <Channel Number="7">Gobo 1</Channel>
   <Channel Number="8">Gobo 2</Channel>
-  <Channel Number="9">Gobo rotation</Channel>
+  <Channel Number="9" ActsOn="8">Gobo 2 Rotation</Channel>
   <Channel Number="10">Prism</Channel>
```
Comment (mcallegari) [inline resources/fixtures/Briteq/Briteq-BT-575S.qxf:170]: What happened here?? An entire mode went away

File: resources/fixtures/American_DJ/American-DJ-Focus-Spot-Three-Z.qxf (hunk 7 above)
[the remark this answers is not in the document]
Comment (mcallegari) [reply resources/fixtures/American_DJ/American-DJ-Focus-Spot-Three-Z.qxf:26]: Those with "rotation" in the description should be rotation presets. Rainbow picture will disappear accordingly.

File: resources/fixtures/American_DJ/American-DJ-Inno-Pocket-Spot-Twins.qxf (hunk 2 above)
Comment (mcallegari) [reply resources/fixtures/American_DJ/American-DJ-Inno-Pocket-Spot-Twins.qxf:17]: Ok, I still don't like it but if we agreed it then let's proceed
I need to check what happen in QLC+ 5 with no picture at all

File: resources/fixtures/American_DJ/American-DJ-Accu-Spot-Pro.qxf (hunk 4 above)
[the remark this answers is not in the document]
Comment (mcallegari) [reply resources/fixtures/American_DJ/American-DJ-Accu-Spot-Pro.qxf:104]: I checked and this was introduced by @kluszon (the one with a thumb up) in 1909885dca31e12f85c194329aa81338830a95a4

File: resources/fixtures/Briteq/Briteq-BT-575S.qxf (hunk 9 above)
Comment (joshuaurieli) [reply resources/fixtures/Briteq/Briteq-BT-575S.qxf:170]: No. The name of the channel was changed (line 70) from "Gobo rotation" to "Gobo 2 Rotation" in order to make it clearer, which Gobo group rotates.

File: resources/fixtures/American_DJ/American-DJ-Focus-Spot-Three-Z.qxf (hunk 7 above)
[the remark this answers is not in the document]
Comment (joshuaurieli) [reply resources/fixtures/American_DJ/American-DJ-Focus-Spot-Three-Z.qxf:26]: OK. We will consistently implement this change.

File: resources/fixtures/American_DJ/American-DJ-Accu-Spot-Pro.qxf (hunk 4 above)
[the remark this answers is not in the document]
Comment (joshuaurieli) [reply resources/fixtures/American_DJ/American-DJ-Accu-Spot-Pro.qxf:104]: OK. We researched this case and confirmed it is PulseSlowToFast.

File: resources/fixtures/American_DJ/American-DJ-Accu-Spot-250-II.qxf (hunk 3 above)
[the remark this answers is not in the document]
Comment (joshuaurieli) [reply resources/fixtures/American_DJ/American-DJ-Accu-Spot-250-II.qxf:33]: What you are suggesting is: remove the GoboRotatingMacro preset **entirely** and just using ActsOn property to define if a gobo is rotating or not. We understand that this is simpler as the user must make one change to define a channel of gobos as rotating. For software engineering and for making fixture definitions closely reflect the device mechanics we think using GoboRotatingMacro and GoboRotatingShakeMakro presets is better. However, simplicity for the user wins over good engineering practice. We will change the fixture definitions as you suggest.

File: resources/fixtures/Briteq/Briteq-BT-575S.qxf (hunk 9 above)
Comment (mcallegari) [reply resources/fixtures/Briteq/Briteq-BT-575S.qxf:170]: Please look closely at the diff.
"8 Channel" mode disappeared and channels from 11 to 15 in "16 channel" mode disappeared.
Basically this definition is now broken.

Comment (joshuaurieli) [reply resources/fixtures/Briteq/Briteq-BT-575S.qxf:170]: Apologies! We will correct it and check for errors like this when preparing future pull requests.

File: resources/fixtures/AFX/AFX-Spot-180-LED.qxf [hunk 10]
```diff
@@ -53,31 +53,31 @@
...
-  <Capability Min="16" Max="31" Preset="GoboMacro" Res1="">Gobo 1</Capability>
-  <Capability Min="32" Max="47" Preset="GoboMacro" Res1="">Gobo 2</Capability>
-  <Capability Min="48" Max="63" Preset="GoboMacro" Res1="">Gobo 3</Capability>
-  <Capability Min="64" Max="79" Preset="GoboMacro" Res1="">Gobo 4</Capability>
-  <Capability Min="80" Max="95" Preset="GoboMacro" Res1="">Gobo 5</Capability>
-  <Capability Min="96" Max="127" Preset="GoboMacro" Res1="">Gobo 6</Capability>
-  <Capability Min="128" Max="191">Gobo Rotate CW slow to fast</Capability>
-  <Capability Min="192" Max="255">Gobo Rotate CCW slow to fast</Capability>
+  <Capability Min="0" Max="15" Preset="GoboMacro" Res1="Other/Open.svg">Open</Capability>
+  <Capability Min="16" Max="31" Preset="GoboMacro">Gobo 1</Capability>
```
Comment (mcallegari) [inline resources/fixtures/AFX/AFX-Spot-180-LED.qxf:63]: I think this should have Res1=""

File: resources/fixtures/American_DJ/American-DJ-Auto-Spot-150.qxf [hunk 11]
```diff
@@ -21,15 +21,15 @@
   <Capability Min="80" Max="95" Preset="GoboMacro" Res1="Others/gobo00032.svg">Gobo 5</Capability>
   <Capability Min="96" Max="111" Preset="GoboMacro" Res1="Robe/gobo00037.svg">Gobo 6</Capability>
   <Capability Min="112" Max="127" Preset="GoboMacro" Res1="SGM/gobo00027.svg">Gobo 7</Capability>
-  <Capability Min="128" Max="255" Preset="GenericPicture" Res1="Others/rainbow.png">Rotating slow-fast</Capability>
+  <Capability Min="128" Max="255" Preset="RotationClockwiseSlowToFast">Rotating slow-fast</Capability>
  </Channel>
  <Channel Name="Gobo Rotation">
   <Group Byte="0">Speed</Group>
```
Comment (mcallegari) [inline resources/fixtures/American_DJ/American-DJ-Auto-Spot-150.qxf:27]: This was 'Gobo' in another definition

File: resources/fixtures/American_DJ/American-DJ-DJ-Spot-300.qxf [hunk 12]
```diff
@@ -35,21 +35,21 @@
...
+  <Capability Min="48" Max="63" Preset="GoboMacro" Res1="">Gobo 3</Capability>
+  <Capability Min="64" Max="79" Preset="GoboMacro" Res1="">Gobo 4</Capability>
+  <Capability Min="80" Max="95" Preset="GoboMacro" Res1="">Gobo 5</Capability>
+  <Capability Min="96" Max="111" Preset="GoboMacro" Res1="">Gobo 6</Capability>
+  <Capability Min="112" Max="127" Preset="GoboMacro" Res1="">Gobo 7</Capability>
+  <Capability Min="128" Max="255" Preset="RotationClockwiseSlowToFast">Gobo scroll</Capability>
  </Channel>
- <Channel Name="Gobo spin">
+ <Channel Name="Gobo Rotation">
   <Group Byte="0">Speed</Group>
```
Comment (mcallegari) [inline resources/fixtures/American_DJ/American-DJ-DJ-Spot-300.qxf:49]: Should be 'Gobo' here?

File: resources/fixtures/Briteq/Briteq-BT-575S.qxf [hunk 13]
```diff
@@ -180,15 +180,15 @@
...
   <Channel Number="6">Prism/Prism rotation/Dimmer</Channel>
   <Channel Number="7">Focus</Channel>
  </Mode>
  <Physical>
-  <Bulb Type="MSD 575W" Lumens="0" ColourTemperature="6500"/>
-  <Dimensions Weight="27.7" Width="427" Height="478" Depth="39"/>
-  <Lens Name="PC" DegreesMin="14" DegreesMax="14"/>
-  <Focus Type="Fixed" PanMax="540" TiltMax="270"/>
-  <Technical PowerConsumption="600" DmxConnector="3-pin"/>
+  <Bulb Type="" Lumens="0" ColourTemperature="0"/>
```
Comment (mcallegari) [inline resources/fixtures/Briteq/Briteq-BT-575S.qxf:188]: All the physical information got lost. Please fix

**Review by mcallegari — CHANGES_REQUESTED** [review]

File: resources/fixtures/Briteq/Briteq-BT-575S.qxf (hunk 13 above)
Comment (joshuaurieli) [reply resources/fixtures/Briteq/Briteq-BT-575S.qxf:188]: Apologies. It will be fixed.

File: resources/fixtures/AFX/AFX-Spot-180-LED.qxf (hunk 10 above)
Comment (joshuaurieli) [reply resources/fixtures/AFX/AFX-Spot-180-LED.qxf:63]: Agreed. Will be done.

File: resources/fixtures/American_DJ/American-DJ-Auto-Spot-150.qxf (hunk 11 above)
Comment (joshuaurieli) [reply resources/fixtures/American_DJ/American-DJ-Auto-Spot-150.qxf:27]: The type of group used for gobo rotation channels varies depending on who created the fixture. Up until now we have considered both "Gobo" or "Speed" as valid. We can make this consistently "Gobo" as well, but then all newly added devices must be checked for consistency. Please advise...

File: resources/fixtures/American_DJ/American-DJ-DJ-Spot-300.qxf (hunk 12 above)
Comment (joshuaurieli) [reply resources/fixtures/American_DJ/American-DJ-DJ-Spot-300.qxf:49]: See comment above

File: resources/fixtures/American_DJ/American-DJ-Auto-Spot-150.qxf (hunk 11 above)
Comment (mcallegari) [reply resources/fixtures/American_DJ/American-DJ-Auto-Spot-150.qxf:27]: Let's stick to "Gobo" as well as "Prism rotation" is set to "Prism"

**Review by mcallegari — CHANGES_REQUESTED** [review]
Please fix the last two "Speed" -> "Gobo" and then this can be merged. Thanks

**Review by mcallegari — APPROVED** [review]

**Pull request merged**